log4j-core/src/main/java/org/apache/logging/log4j/core/config/AbstractConfiguration.java

```diff
@@ -323,10 +325,19 @@ public void start() {
         LOGGER.info("Starting configuration {}...", this);
         this.setStarting();
         if (watchManager.getIntervalSeconds() >= 0) {
-            LOGGER.info(
-                    "Start watching for changes to {} every {} seconds",
-                    getConfigurationSource(),
-                    watchManager.getIntervalSeconds());
+            if (uris != null && uris.size() > 0) {
+                LOGGER.info(
+                        "Start watching for changes to {} and {} every {} seconds",
+                        getConfigurationSource(),
+                        uris,
+                        watchManager.getIntervalSeconds());
+                watchMonitorUris();
+            } else {
+                LOGGER.info(
+                        "Start watching for changes to {} every {} seconds",
+                        getConfigurationSource(),
+                        watchManager.getIntervalSeconds());
+            }
```

Comment:
Can't we simplify this as follows? (Can `uris` be null at all?)

```suggestion
            LOGGER.info(
                    "Start watching for changes to {} and {} every {} seconds",
                    getConfigurationSource(),
                    uris,
                    watchManager.getIntervalSeconds());
            watchMonitorUris();
```

Reply:
... and, this is the wrong place and time to do admit monitor URIs to the `WatchManager`. Would you move this admission operation (i.e., the logic in `watchMonitorUris`) to `initializeWatchers`, please?

Reply:
Hi @vy 
The reason I did not put it in initializeWatchers() is because initializeWatchers() is called before the elements are read. The top level attributes (including monitorInterval) have been read at this stage and can therefore be used, but the elements (including monitorUris) have not been.

Would it be ok to move invoking watchMonitorUris() to intialization() which is earlier than here, but after the elements have been read? This would avoid having to read the monitorUri element earlier and separately from the other elements.

I can move it to initializeWatchers() either if you still prefer that, and add code to read the monitorUri element separately to the other elements, but I think the above would be cleaner in the code.